### DISCUSSION TITLE
Failed to parse service type, ignoring: exec

### DISCUSSION THREAD

**Discussion by tekumara** [coder/code-server#3687] [Q&A]
```
$ code-server --version
3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005

$ sudo systemctl status code-server@ubuntu.service | cat
● code-server@ubuntu.service - code-server
   Loaded: loaded (/usr/lib/systemd/system/code-server@.service; indirect; vendor preset: enabled)
   Active: inactive (dead)

Jun 25 06:51:31 ip-10-97-36-177 systemd[1]: /usr/lib/systemd/system/code-server@.service:6: Failed to parse service type, ignoring: exec
Jun 25 06:51:41 ip-10-97-36-177 systemd[1]: /usr/lib/systemd/system/code-server@.service:6: Failed to parse service type, ignoring: exec

$ cat /usr/lib/systemd/system/code-server@.service
[Unit]
Description=code-server
After=network.target

[Service]
Type=exec
ExecStart=/usr/bin/code-server
Restart=always
User=%i

[Install]
WantedBy=default.target
```

**Comment by tekumara** [comment]
code-server will start regardless, but maybe `Type=exec` can be removed so this warning message doesn't appear?

**Reply by jsjoeio** [reply]
Is this an issue with how the service is set up in your environment or something specific to code-server? (apologies, not super familiar with systemd services)

**Reply by tekumara** [reply]
_/usr/lib/systemd/system/code-server@.service_ as above is shipped in the [code-server .deb package](https://github.com/cdr/code-server/releases/download/v3.10.2/code-server_3.10.2_amd64.deb), so this is possibly something to do with how code-server is packaged

**Reply by tekumara** [reply]
See
* https://github.com/cdr/code-server/blob/main/ci/build/code-server-user.service#L6 
* https://github.com/cdr/code-server/blob/main/ci/build/code-server%40.service#L6

**Reply by jsjoeio** [reply]
Ah, sorry I misunderstand. Thanks for the links! cc @oxy who knows more about this part of the codebase
